crates/bevy_ecs/src/system/query.rs

```diff
@@ -10,235 +10,265 @@ use crate::{
-/// # Usage of query results
+/// An alternative to this idiom is to wrap the conflicting queries into a [`ParamSet`](super::ParamSet).
 ///
-/// Inside the body of the system function, the `Query` is available as a function parameter.
-/// This section shows various methods to access query results.
+/// # Accessing query items
 ///
-/// ## Iteration over every query result
+/// The following table summarizes the behavior of the safe methods that can be used to get query items.
 ///
-/// The [`iter`](Self::iter) and [`iter_mut`](Self::iter_mut) methods are used to iterate
-/// over every query result. Refer to the
-/// [`Iterator` API docs](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html)
-/// for advanced iterator usage.
+/// |Query methods|Effect|
+/// |:---:|---|
+/// |[`iter`]\([`_mut`][iter_mut])|Returns an iterator over all query items.|
+/// |[`for_each`]\([`_mut`][for_each_mut]),<br>[`par_for_each`]\([`_mut`][par_for_each_mut])|Runs a specified function for each query item.|
+/// |[`iter_many`],<br>[`many_for_each_mut`]|Iterates or runs a specified function over query items generated by a list of entities.|
+/// |[`iter_combinations`]\([`_mut`][iter_combinations_mut])|Returns an iterator over all combinations of a specified number of query items.|
+/// |[`get`]\([`_mut`][get_mut])|Returns the query item for the specified entity.|
+/// |[`many`]\([`_mut`][many_mut]),<br>[`get_many`]\([`_mut`][get_many_mut])|Returns the query items for the specified entities.|
+/// |[`single`]\([`_mut`][single_mut]),<br>[`get_single`]\([`_mut`][get_single_mut])|Returns the query item while veryfing that there aren't others.|
 ///
-/// ```
-/// # use bevy_ecs::prelude::*;
-/// # #[derive(Component)]
-/// # struct ComponentA;
-/// # #[derive(Component)]
-/// # struct ComponentB;
-/// fn immutable_query_system(query: Query<(&ComponentA, &ComponentB)>) {
-///     for (a, b) in query.iter() {
-///         // Here, `a` and `b` are normal references to components, relatively of
-///         // `&ComponentA` and `&ComponentB` types.
-///     }
-/// }
-/// # bevy_ecs::system::assert_is_system(immutable_query_system);
+/// There are two methods for each type of query operation: immutable and mutable (ending with `_mut`).
+/// When using immutable methods, the query items returned are of type [`ROQueryItem`], a read-only version of the query item.
+/// In this circumstance, every mutable reference in the query fetch type parameter is substituted by a shared reference.
 ///
-/// fn mutable_query_system(mut query: Query<(&mut ComponentA, &ComponentB)>) {
-///     for (mut a, b) in query.iter_mut() {
-///         // Similar to the above system, but this time `ComponentA` can be accessed mutably.
-///         // Note the usage of `mut` in the tuple and the call to `iter_mut` instead of `iter`.
-///     }
-/// }
-/// # bevy_ecs::system::assert_is_system(mutable_query_system);
-/// ```
+/// # Performance
+///
+/// Creating a `Query` is a low-cost operation.
```

Comment:
```suggestion
/// Creating a `Query` is a `O(1)` operation.
```

Alternatively, low-cost *constant* operation.

Reply:
I went for the wordy alternative.